Users/binah/access-control/access-control.cs

```diff
@@ -0,0 +1,320 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Sansar.Script;
+using Sansar.Simulation;
+using System.Diagnostics;
+
+// register agents on entry
+
+public class AccessControl : SceneObjectScript
+{
+    #region EditorProperties
+    [Tooltip("Comma separated list of user handles for users that are allowed to ban other users.\nThe scene owner is always an admin and does not need to be on this list.")]
+    [DisplayName("Admin Handles")]
+    public string AdminHandles;
+
+    [DefaultValue(true)]
+    [Tooltip("Allow anyone to enter without an access check")]
+    [DisplayName("DoorsOpen")]
+    public bool DoorsOpen;
+
+    [DefaultValue("/showlog")]
+    [Tooltip("Shows visitor log, door open state and banlist")]
+    [DisplayName("Chat Command")]
+    public readonly string VisitorListCommand = "/showlog";
+
+    [DefaultValue("/toggleDoor")]
+    [Tooltip("Toggle door open state")]
+    [DisplayName("Chat Command")]
+    public readonly string ToggleDoorCommand = "/toggleDoor";
+
+    [DefaultValue("/ban")]
+    [Tooltip("Command to ban a user")]
+    [DisplayName("Chat Command")]
+    public readonly string BanCommand = "/ban";
+
+    [DefaultValue("/unban")]
+    [Tooltip("Command to remove a user from banlist")]
+    [DisplayName("Chat Command")]
+    public readonly string UnBanCommand = "/unban";
+
+    [DisplayName("Banned Destination")]
+    [Tooltip("The destination to send banned users.")]
+    [DefaultValue("https://atlas.sansar.com/experiences/katylina/you-have-been-expelled")]
+    public string BannedDestination;
+
+    [DefaultValue(true)]
+    public bool DebugLogging;
+    #endregion
+
+    private RigidBodyComponent _rb;
+
+    private List<string> Admins = new List<string>(); // handle
+    private List<string> Banned = new List<string>(); 
+
+    public override void Init()
+    {
+        // get the holding container rigid body
+        if (!ObjectPrivate.TryGetFirstComponent(out _rb))
+        {
+            if (DebugLogging) Log.Write("Script not running on an object with a physics volume!");
```

Comment:
You should probably return here if the script can't function without a rigidbody. If that is the case then this should not be an optional log, its an error for all intents and purposes. I usually use `LogLevel.Warning` because its not a runtime error but still important enough that the user needs to see it in log.